### PR TITLE
feat: Add react-components lib

### PR DESCRIPTION
**Pull request by eileenmguo**
Moving the react-component lib from kadena-io/javascript-monorepo to this repo

## Check off the following:

- [x] I have reviewed my changes and run the appropriate tests.
- [x] I have have run `rush change` to add the appropriate change logs.
- [ ] I have added/edited docs.
- [ ] I have added tutorials.
- [ ] I have double checked and DEFINITELY added docs.

### CODE REVIEW

**Review by sstraatemans — APPROVED** [review]

File: packages/libs/react-components/src/styles/baseGlobalStyles.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
+// NOTE: Refer to https://courses.joshwcomeau.com/css-for-js/treasure-trove/010-global-styles for more detailed explanation
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const baseGlobalStyles: Record<string, any> = {
```
Comment (alber70g) [inline packages/libs/react-components/src/styles/baseGlobalStyles.ts:4]: Can we use `unknown` here?

File: packages/libs/react-components/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "extends": "./node_modules/@kadena-dev/heft-rig/profiles/default/tsconfig-base.json",
+  "compilerOptions": {
+    "types": ["heft-jest"],
+    "target": "es5",
```
Comment (alber70g) [inline packages/libs/react-components/tsconfig.json:5]: Can we use a more modern version, or are we leaving consumers behind. I'd argue that when you're using the `esm` variant, you will have a modern browser as well. Based on this table we could find one that's widely used for ESM variant: https://kangax.github.io/compat-table/es2016plus/ (ignoring IE11 of course)
Maybe a minimum of ES2017 would be doable.

File: packages/libs/react-components/package.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    "@rushstack/eslint-config": "~2.6.2",
+    "@rushstack/heft": "~0.46.1",
+    "@storybook/addon-actions": "^7.0.0-rc.3",
+    "@storybook/addon-essentials": "^7.0.0-rc.3",
+    "@storybook/addon-interactions": "^7.0.0-rc.3",
+    "@storybook/addon-links": "^7.0.0-rc.3",
+    "@storybook/addon-mdx-gfm": "^7.0.0-rc.3",
+    "@storybook/react": "^7.0.0-rc.3",
+    "@storybook/react-webpack5": "^7.0.0-rc.3",
+    "@storybook/testing-library": "^0.0.14-next.1",
```
[the remark this answers is not in the document]
Comment (eileenmguo) [reply packages/libs/react-components/package.json:49]: no specific reasons, these were the packages that were added using storybook's command line tool `npx storybook init`

File: packages/libs/react-components/src/styles/baseGlobalStyles.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
+// NOTE: Refer to https://courses.joshwcomeau.com/css-for-js/treasure-trove/010-global-styles for more detailed explanation
```
[the remark this answers is not in the document]
Comment (eileenmguo) [reply packages/libs/react-components/src/styles/baseGlobalStyles.ts:1]: I've updated it with a link to his public article about this!

File: packages/libs/react-components/src/styles/colors/index.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  background: 'Gainsboro',
+  fontColor: 'Gainsboro',
+  gray500: 'DarkSlateGray',
+  blue500: 'SteelBlue',
+  purple500: 'RebeccaPurple',
+  green500: 'DarkGreen',
+  red500: 'DarkRed',
+};
+
+export const darkColors: Record<string, string> = {
```
[the remark this answers is not in the document]
Comment (eileenmguo) [reply packages/libs/react-components/src/styles/colors/index.ts:12]: No preference here for me, I'll change it 👍

File: packages/libs/react-components/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (alber70g) [reply packages/libs/react-components/package.json:49]: Then that'll be good enough if their cli tool has them as defaults

**Pull request merged**